components/account/UserButtons.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div
+    class="user-buttons"
+    data-qa="user buttons"
+  >
+    <b-button
+      class="icon-ic-add"
+      data-qa="add to gallery button"
+      :aria-label="$t('actions.addToGallery')"
+    />
+    <b-button
+      :pressed="liked"
+      class="icon-heart"
+      data-qa="like button"
+      :aria-label="$t('actions.like')"
+      size="sm"
+      @click="doLike()"
+    />
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'UserButtons',
+
+    props: {
+      itemUrl: {
+        type: [String, Object],
+        default: ''
+      }
+    },
+
+    data() {
+      return {
+        liked: false
+      };
+    },
+    computed: {
+      itemId() {
+        return this.itemUrl.params[0];
```

Comment:
This presumes that the itemUrl is a route object, which is fine in the scenarios we have for the user buttons at present, but means that the prop should only accept object-type values, not strings.